nvflare/app_common/workflows/scatter_and_gather_autofedrl.py

```diff
@@ -0,0 +1,477 @@
+        if not isinstance(self.search_space, FLComponent):
+            self.system_panic(
+                f"Shareable generator {self.search_space_id} must be a ShareableGenerator type object, "
+                f"but got {type(self.search_space)}",
+                fl_ctx,
+            )
+            return
+
+        self.persistor = self._engine.get_component(self.persistor_id)
+        if not isinstance(self.persistor, LearnablePersistor):
+            self.system_panic(
+                f"Model Persistor {self.persistor_id} must be a LearnablePersistor type object, "
+                f"but got {type(self.persistor)}",
+                fl_ctx,
+            )
+            return
+
+        # initialize global model
+        fl_ctx.set_prop(AppConstants.START_ROUND, self._start_round, private=True, sticky=True)
+        fl_ctx.set_prop(AppConstants.NUM_ROUNDS, self._num_rounds, private=True, sticky=False)
+        self._global_weights = self.persistor.load(fl_ctx)
+        fl_ctx.set_prop(AppConstants.GLOBAL_MODEL, self._global_weights, private=True, sticky=True)
+        self.fire_event(AppEventType.INITIAL_MODEL_LOADED, fl_ctx)
+
+    def control_flow(self, abort_signal: Signal, fl_ctx: FLContext) -> None:
+        try:
+
+            self.log_info(fl_ctx, "Beginning ScatterAndGather training phase.")
+            self._phase = AppConstants.PHASE_TRAIN
+
+            fl_ctx.set_prop(AppConstants.PHASE, self._phase, private=True, sticky=False)
+            fl_ctx.set_prop(AppConstants.NUM_ROUNDS, self._num_rounds, private=True, sticky=False)
+            self.fire_event(AppEventType.TRAINING_STARTED, fl_ctx)
+
+            if self._current_round is None:
+                self._current_round = self._start_round
+            while self._current_round < self._start_round + self._num_rounds:
+
+                if self._check_abort_signal(fl_ctx, abort_signal):
+                    return
+
+                self.log_info(fl_ctx, f"Round {self._current_round} started.")
+                fl_ctx.set_prop(AppConstants.GLOBAL_MODEL, self._global_weights, private=True, sticky=True)
+                fl_ctx.set_prop(AppConstants.CURRENT_ROUND, self._current_round, private=True, sticky=False)
+                self.fire_event(AppEventType.ROUND_STARTED, fl_ctx)
+
+                # Search hyperparameters
+                self.search_space.sample_hyperparamters(fl_ctx)
+
+                # Create val_task for hp search
+                data_shareable_val: Shareable = self.shareable_gen.learnable_to_shareable(self._global_weights, fl_ctx)
+                data_shareable_val.set_header(AppConstants.CURRENT_ROUND, self._current_round)
+                data_shareable_val.set_header(AppConstants.NUM_ROUNDS, self._num_rounds)
+                data_shareable_val.add_cookie(AppConstants.CONTRIBUTION_ROUND, self._current_round)
+
+                val_task = Task(
+                    name="model_validate_for_hp_search",
```

Comment:
this name should use the constants that we defined.

Reply:
Yeah, changed the task name to defined constant. https://github.com/NVIDIA/NVFlare/pull/1283/commits/620afaa85194b1d37e7688cb4fc15a069dd529af